drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c

```diff
@@ -201,12 +204,17 @@ static void submac_tx_done(ieee802154_submac_t *submac, int status,
                            ieee802154_tx_info_t *info)
 {
     (void)status;
-    (void)info;
     netdev_ieee802154_submac_t *netdev_submac = container_of(submac,
                                                              netdev_ieee802154_submac_t,
                                                              submac);
     netdev_t *netdev = (netdev_t *)netdev_submac;
 
+    if (info) {
+        netdev_submac->retrans = info->retrans;
+    } else {
+        netdev_submac->retrans = -1;
+    }
+
```

Comment:
Since you are there, could you add this patch?
```
diff --git a/sys/net/link_layer/ieee802154/submac.c b/sys/net/link_layer/ieee802154/submac.c
index 0584ba95a3..392714b995 100644
--- a/sys/net/link_layer/ieee802154/submac.c
+++ b/sys/net/link_layer/ieee802154/submac.c
@@ -194,6 +194,9 @@ static void _handle_tx_success(ieee802154_submac_t *submac,
 
     if (ieee802154_radio_has_frame_retrans(dev) ||
         ieee802154_radio_has_irq_ack_timeout(dev) || !submac->wait_for_ack) {
+        if (!ieee802154_radio_has_frame_retrans_info()) {
+            info->retrans = -1;
+        }
         _tx_end(submac, info->status, info);
     }
     else {
```

And change these lines to:
```suggestion
    if (info) {
        netdev_submac->retrans = info->retrans;
    }
```
?
That way the SubMAC will always give its best effort to report the retransmissions. And if there's no info (TX_MEDIUM_BUSY or TX_NO_ACK), then `netdev_submac->retrans` doesn't get updated.

Reply:
I think turning the `_get_cap()` function into a bitfield would be a good optimization.

Reply:
yes, we could!